state/state.go

```diff
@@ -1468,54 +1507,31 @@ func (st *State) processIAASModelApplicationArgs(args *AddApplicationArgs) error
 					err, "cannot deploy to machine %s", m,
 				)
 			}
+			// We've checked the machine placement, there's not a
+			// provider-specific placement that needs checking
+			// (although the instance creation still needs to be
+			// checked).
+			placements.Add("")
```

Comment:
This seems a bit icky. Shouldn't the check just skip over non-specified placement

Reply:
No, we should check the arguments as if there is no provider-relevant placement, so the provider gets a chance to check constraints and volumes.

Reply:
Oh, do you mean we don't need to do this unless there're no other placement directives so we can just rely on the check at the end?

Reply:
Ah, I think I understand your point now - for machine-placement, we're specifying an existing machine, so we don't need to check instance args because we're not going to create a new instance. Is that right?
That solves the problem with the manual provider as well - we should always be using machine placement to deploy in that case because the cloud can't create instances.